src/game/board/boardSituation.js

```diff
@@ -1,14 +1,31 @@
-export default class BoardSituation {
-    constructor() {
+export default class BoardSituation {
+        constructor(firstBoardPosition) {
```

Comment:
Z tym konstruktorem jest coś mocno nie tak. Po pierwsze, newBoardPosition chyba nigdzie nie zoostaje zadeklarowane, więc zakładam, że argument konstruktora i to, co wstawiasz do this.boardPosition też miały być się newBoardPosition nazywać. Ale nawet wtedy 
 - nie ma możliwości stworzenia nowej sytuacji, która miałaby określoną poprzednią sytuację
 - poprzednia sytuacja jest ustawiana jako tablica, co NIGDY nie powinno mieć miejsca. Tam powinien być albo obiekt klasy BoardSituation albo null.